### DISCUSSION TITLE
Can't make App Check work with Firestore on iOS

### DISCUSSION THREAD

**Discussion by anhnch** [invertase/react-native-firebase#6021] [Q&A] [answered]
Hi,

I've just included App Check to my RN app. I've done all these steps
- Added the package @react-native-firebase/app-check
- set platform :ios, '11.0' in Podfile
- Created a Key in Apple Developer Console with DeviceCheck and App Attest
- Enabled DeviceCheck in Firebase console Project settings with Key ID and Team ID
- I also App Attest in Firebase console 
- Enforced app check for Firestore
- Add the capability App Attest in XCode Signing & Capabilities
- Built and ran the app

But I got this 
```
[Error: [firestore/permission-denied] The caller does not have permission to execute the specified operation.]
```
and a lot of "Unverified: invalid requests" from the Firebase Console's App Check request metrics chart.

I'm testing App check with Firestore on a real iPhone device in debug and release mode.  Activating like below doesn't work too.
```typescript
await firebase.appCheck().activate('ignored', true); // is this really necessary?
```

Do I miss something?

**Comment by anhnch** [comment, marked as answer]
[SOLVED]
When I looked into the XCode logs I found this
```
[Firebase/Firestore][I-FST000001] AppCheck failed: 'The operation couldn’t be completed. The server responded with an error: 
 - URL: https://firebaseappcheck.googleapis.com/v1beta/projects/<my-project>/apps/1:<my-app-id>:ios:<id>:exchangeDeviceCheckToken 
 - HTTP status code: 403 
 - Response body: {
  "error": {
    "code": 403,
    "message": "App attestation failed.",
    "status": "PERMISSION_DENIED"
  }
}
```
I went to DeviceCheck settings in the Firebase console, checked the Auth key and found that I uploaded the wrong p8 file.

I went back to the Apple Certificates, Identifier & Profile and I downloaded the key, then uploaded it to the Firebase console. It works fine now. Hope this is helpful to someone.

**Comment by nixolas1** [comment]
I had a similar issue, but my problem was that I had used the wrong Team-ID. So to anyone else getting the same error response, check your firebase config :)

**Comment by OmarKhairyMo** [comment]
We have an app live during the smoke testing we havent faced this issue at all, but in our Error monitoring we found a lot of users facing this issue. My question is: if i used the wrong p8 file should all the users face the same issue not part of it right ?

**Comment by tarkcelk** [comment]
@OmarKhairyMo facing the same issue

**Comment by abgaryanharutyun** [comment]
The same issue on production build im getting lot of users where don't have token, also on dev mode getting 403 error

**Comment by glennposadas** [comment]
This is driving me nuts, tbh. Lol. A lot of failure. Real device or not, I get this kind of message.
I've checked everything, from team Id, keys, names, entitlements, config, hierarchy of firebase config calls, and whatnot. I'm thinking of ditching this, idk. Kind of pain in the ass, it blocks me from moving forward.

**Comment by SaadFarooq-Dev** [comment]
@glennposadas Did you found any solution for it? Thanks.

**Reply by glennposadas** [reply]
@SaadFarooq-Dev I ditched this.

**Reply by SaadFarooq-Dev** [reply]
@glennposadas Did you use any alternative for it?

**Comment by Fortidude** [comment]
Anyone interested - I spend few Saturdays on this topic. tried all kind of different library versions, apple keys etc

nothing works. I ditched it. Garbage trash.
